### PR TITLE
add user parsing in HDFS URI

### PR DESCRIPTION
**Pull request by akonyaev90**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

- Improvement
Short description :
add user parsing in HDFS engine builder

Detailed description :
when we using HDFS with hadoop.security.authentication=simple, we can't set username for hdfs-table in ClickHouse.
hdfsLib use user from under which clickhouse is running
User in URI is ignored
I think this must be fixed

### CODE REVIEW

File: dbms/src/IO/HDFSCommon.cpp [hunk 1]
```diff
@@ -25,6 +26,13 @@ HDFSBuilderPtr createHDFSBuilder(const Poco::URI & uri)
     hdfsBuilderConfSetStr(builder.get(), "input.write.timeout", "60000"); // 1 min
     hdfsBuilderConfSetStr(builder.get(), "input.connect.timeout", "60000"); // 1 min
 
+    std::string userInfo = uri.getUserInfo();
+    if (!userInfo.empty() && userInfo.front() != ':') {
+        char *user = strtok(const_cast<char *> (userInfo.c_str()), ":");
```
Comment (alexey-milovidov) [inline dbms/src/IO/HDFSCommon.cpp]: Need to rewrite in C++.
PS. strtok can damage password field if it contains colon.

Comment (alexey-milovidov) [reply dbms/src/IO/HDFSCommon.cpp]:
```
BUGS
       * The strtok() function uses a static buffer while parsing, so it's not thread safe.  Use strtok_r() if this matters to you.
```

Comment (akonyaev90) [reply dbms/src/IO/HDFSCommon.cpp]: Rewrite in c++ 
Used  boost::split like in other code examples in repo.
Fixed code styles bugs
NULL -> nullptr for fix clang builds

Comment (alexey-milovidov) [reply dbms/src/IO/HDFSCommon.cpp]: You can just use std::string::find.

Comment (akonyaev90) [reply dbms/src/IO/HDFSCommon.cpp]: rewritten from boost to std::string

**Comment by akonyaev90** [comment]
I have no idea why my commit fail this test - '00908_long_http_insert' (((

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
> I have no idea why my commit fail this test - '00908_long_http_insert' (((

It's unrelated.

**Pull request merged**